### PR TITLE
Adding new document about migration and dockerize osclass.

### PR DESCRIPTION
**Pull request by XaviOutside**
Signed-off-by: XaviOutside <xavioutside@hotmail.com>

### CODE REVIEW

File: docs/osclass.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+1. Create the virtualbox machine.
+
+        $ docker-machine create --driver virtualbox dockerenv 
+
+        $ docker-machine active dockerenv
+
+        $ $(docker-machine env dockerenv)
+
+1. Download the scripts using `git clone`.
```
Comment (thaJeztah) [inline docs/osclass.md]: Could you change the numbers in the numbered list to the actual numbers? I know it's not a requirement for Markdown, but it's confusing when reading the raw source.

File: docs/osclass.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+        $ git clone https://github.com/XaviOutside/MIGRATION_OSCLASS_2_DOCKER.git
+        
+1. Change to the folder directory.
+
+        $ cd MIGRATION_OSCLASS_2_DOCKER
+
+1. Open the  in `common.env` file in your favorite editor.
+
+        $ vi common.env
```
Comment (thaJeztah) [inline docs/osclass.md]: Perhaps we should just remove this line and leave it up to the user what to use here (same as there's no example below "save and close the file")?

**Comment by thaJeztah** [comment]
Two nits, other than that, LGTM. Thanks!


ping @fredlf @jamtur01 @SvenDowideit @moxiegirl

**Comment by moxiegirl** [comment]
@XaviOutside Let me take a look and Gist you some updates.  Some ESL boo-boos in this.

Glad you are settled in Germany and back contributing.  I loved my time in Germany but ahhh Spain...still need to visit there.  :-D

File: docs/osclass.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
+<!--[metadata]>
```
Comment (moxiegirl) [inline docs/osclass.md]: @XaviOutside  thank you for this.  It isn't really getting started as much as migrating.  So, let's make that clear in the title.  I did a quick and dirty overhaul and add a callout to you to add some info:

-->https://gist.github.com/moxiegirl/7a51734a4bbc23935dd3




# Migrate Osclass installation to Docker Compose

This document teaches you how to migrate an Osclass installation to a Docker
infrastructure. When you are done, your Docker infrastructure will contain six
Docker containers, three server containers, and three volume containers:

* Mysql database
* Postfix server
* Apache2 Webserver
* Osclass volume 
* Database volume
* Backup volume

This configuration was tested on Osclass 3.5.

## Prerequisites

Before you begin, make sure you have installed the following software:

* VirtualBox 
* [Docker](https://docs.docker.com/installation/#installation)
* [Docker Machine](http://docs.docker.com/machine/install-machine/)
* [Docker Compose](install.md)

Refer to the appropriate Docker documentation for different distributions for installing.

## Provision the servers with machine

XAVIER: Add an intro sentence or two about how this procedure users yoru scripts and what they are doing

1. Create the VirtualBox machine.

        $ docker-machine create --driver virtualbox dockerenv 

        $ docker-machine active dockerenv

        $ $(docker-machine env dockerenv)

2. Download the scripts using `git clone`.

        $ git clone https://github.com/XaviOutside/MIGRATION_OSCLASS_2_DOCKER.git
        
3. Change to the folder directory.

        $ cd MIGRATION_OSCLASS_2_DOCKER

4. Open the `common.env` file in your favorite editor, for example:

        $ vi common.env 

5. Set the `DOMAIN` and `MYSQL_ROOT_PASSWORD` to contain your domain and password.

         DOMAIN=yourdomain.org
         MYSQL_ROOT_PASSWORD=yourpassword

6. Save and close the file.

7. Build containers.

        $ docker-compose build

8. Run containers.

        $ docker-compose up

9. Launch the `http://your_ip_docker_machine` URL in your favorite browses.

   Use the following command to get your `ip` with machine:
   
        $ docker-machine ip
     

You can also use the `INSTALL.sh` file if you want to launch without `docker-compose`.


## Backup your Osclass server and import the data

1. Backup to a `tar.gz` file of your current Osclass server.

2. Replace the `backup_osclass.tar.gz` in the `migration` folder.
     
3. Create a `backup.mysql.sql` dump file of your Osclass database.

4. Replace the `backup.mysql.sql` in the `migration` folder.
     
5. Copy the `migration` folder into the Osclass container.

        $ tar -cf - migration | docker exec -i <NAME_OSCLASS_CONTAINER>  /bin/tar -C / -xf -

6. Run the `osclass_init.sh` script to import the files and database data into the volume containers.

        $ docker exec -it <NAME_OSCLASS_CONTAINER> bash /osclass_init.sh

7. Generate a backup of the Osclass files and database.

        $ docker exec -it <NAME_OSCLASS_CONTAINER> bash /osclass_backup.sh


## More Compose documentation

- [User guide](/)
- [Installing Compose](install.md)
- [Get started with Django](django.md)
- [Get started with Rails](rails.md)
- [Get started with Wordpress](wordpress.md)
- [Migration: Dockerize osclass](osclass.md)
- [Command line reference](cli.md)
- [Yaml file reference](yml.md)
- [Compose environment variables](env.md)
- [Compose command line completion](completion.md)

**Comment by XaviOutside** [comment]
Hi,

I don't know if I have to do something else but I made some changes after your feedbacks and comments and I would like to know if its ready to merge or not. Thanks!

**Comment by moxiegirl** [comment]
LGTM thanks @XaviOutside

**Comment by moxiegirl** [comment]
@XaviOutside I want to make sure that the add is good for @aanand --- if he is happy we will merge. He is the component owner.

**Comment by aanand** [comment]
Sorry I haven't paid much attention to this. In general, the more guides the merrier, as long as they're for popular software.

However, I've never even heard of Osclass before, and I don't know that it's a popular enough piece of software to merit a page in Compose's official documentation. (This is a very rough measure, but it [doesn't register on Google Trends](https://www.google.com/trends/explore#q=docker%20rails%2C%20docker%20django%2C%20docker%20wordpress%2C%20docker%20node%2C%20docker%20osclass&cmpt=q&tz=Etc%2FGMT-1) for things people want to dockerize compared to Rails, Django, Node or Wordpress.)

For that reason, while it could e.g. make a great blog post, I don't think it belongs in the official Compose docs.

**Comment by dnephin** [comment]
I agree with @aanand

This has come up before, and I think it would be good to have a wiki page with links to a lot more examples. We can link to that wiki from the official documentation, so it still gives developers a chance to easily discover all of these examples without having to update the core documentation.

**Comment by moxiegirl** [comment]
@dnephin It might be best to put the examples page in the official documentation rather than a GitHub wiki.  We seem to be reserving the wiki for planning. I'm happy to create an examples section. @XaviOutside I'm going to close this. Let me know if you put up a blog we can link to.

**Comment by dnephin** [comment]
@moxiegirl sounds good. I created #1951

**Comment by XaviOutside** [comment]
Thanks @moxiegirl and @dnephin! I have put the doc in this url http://blog.nayubiko.com/2015/09/how-to-migrate-osclass-to-docker-and.html <http://blog.nayubiko.com/2015/09/how-to-migrate-osclass-to-docker-and.html>
This is the blog of my web site.

**Pull request closed without merging**